### ISSUE TITLE
Failure debug mode breaks Deferred/Failure integration

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5089]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5089|
|Type|release blocker: regression|
|Created|2011-05-06 16:31:03Z|
|Branch|https://github.com/twisted/twisted/tree/failure-debug-5089|

```
>>> from twisted.python.failure import startDebugMode
/System/Library/Frameworks/Python.framework/Versions/2.6/Extras/lib/python/zope/__init__.py:1: UserWarning: Module twisted was already imported from twisted/__init__.pyc, but /System/Library/Frameworks/Python.framework/Versions/2.6/Extras/lib/python is being added to sys.path
  __import__('pkg_resources').declare_namespace(__name__)
>>> startDebugMode()
>>> from twisted.internet.defer import maybeDeferred
>>> maybeDeferred(lambda: 1/0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "twisted/internet/defer.py", line 135, in maybeDeferred
    return fail(failure.Failure(captureVars=Deferred.debug))
TypeError: _debuginit() got an unexpected keyword argument 'captureVars'
>>> 
```

<details><summary>Searchable metadata</summary>

```
trac-id__5089 5089
type__release_blocker__regression release blocker: regression
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_failure_debug_5089 branches/failure-debug-5089
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1304699463000000 1304699463000000
changetime__1305118479000000 1305118479000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

This was introduced by [#5011](https://github.com/twisted/twisted/issues/5011).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @itamarst<br>@itamarst set status to `assigned`|
|-|-|

Easy fix, just need to do some paperwork...

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

(In [31788]) Branching to 'failure-debug-5089'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst removed owner<br>@itamarst set status to `new`|
|-|-|

OK, ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @itamarst|
|-|-|

Hooray, thanks.

  1. It's a trivial difference, but `Failure.__dict__['__init__']` will be a function before these tests run and an unbound method afterwards.
  1. It might be nice to assert that variables are or are not captured in these cases, as determined by the value passed for `captureVars`.
  1. There's some trailing whitespace in `test_captureVars`
  1. Maybe `M-x set-fill-column 78` and reflow the docstrings, if you feel like it :).  Also, don't say "should".
  1. Add a .misc fragment

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_7"></a>|@itamarst removed owner|
|-|-|

OK, fixed all of those.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set owner to @itamarst|
|-|-|

.misc files don't need any content (and their content won't go into the news file).  I suggested .misc instead of .bugfix because this is an unreleased regression; I don't think people reading the release notes will care about fixes for such things.  Otherwise great, [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/failure-debug-5089) are good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_9"></a>|@itamarst set status to `closed`|
|-|-|

(In [31807]) Merge failure-debug-5089.

Author: itamar
Reviewer: exarkun
Fixes: [#5089](https://github.com/twisted/twisted/issues/5089)

Fix regression that broke t.p.failure.startDebugMode() and therefore 'trial -b'.
Also, added tests to prevent this from breaking in future.
